tests/runfiles/sunos.run

```diff
@@ -37,3 +37,7 @@ tags = ['functional', 'cli_root', 'zpool_create']
 tests = ['zvol_misc_001_neg', 'zvol_misc_003_neg', 'zvol_misc_004_pos',
     'zvol_misc_005_neg', 'zvol_misc_006_pos']
 tags = ['functional', 'zvol', 'zvol_misc']
+
+[tests/functional/zvol/zvol_swap:illumos]
+tests = ['zvol_swap_003_pos', 'zvol_swap_005_pos', 'zvol_swap_006_pos']
+tags = ['functional', 'zvol', 'zvol_swap']
```

Comment:
Making this illumos specific means we can remove them from the known skip list in `tests/test-runner/bin/zts-report.py`, and drop the `if is_linux || is_freebsd` check at the top of each test case.